libraries/AP_Compass/Compass_learn.cpp

```diff
@@ -164,7 +164,9 @@ void CompassLearn::update(void)
             sample_available = false;
             num_samples = 0;
             have_earth_field = false;
-            memset(predicted_offsets, 0, sizeof(predicted_offsets));
```

Comment:
I am not expert on asm but the gcc explorer output with memset was much simpler. I close the pragma solution as it wasn't maintainable simply with different compilers versions. The zeroing isn't called often here so that if fine and at least that is a cleaner solution than memset !